plugins/woocommerce/includes/wc-update-functions.php

```diff
@@ -2730,6 +2730,39 @@ function wc_update_910_add_launch_your_store_tour_option() {
 	add_option( 'woocommerce_show_lys_tour', 'yes' );
 }
 
+/**
+ * Add woocommerce_hooked_blocks_version option for existing stores that are using a theme that supports the Block Hooks API
+ */
+function wc_update_920_add_wc_hooked_blocks_version_option() {
+	if ( ! wc_current_theme_is_fse_theme() && ! current_theme_supports( 'block-template-parts' ) ) {
+		return;
+	}
+
+	$option_name       = 'woocommerce_hooked_blocks_version';
+	$option_value      = get_option( $option_name );
+	$active_theme_name = wp_get_theme()->get( 'Name' );
+	/**
+	 * A list of theme slugs to execute this with.
+	 * We are applying this filter to allow for the list to be extended by third-parties who were already using it.
+	 *
+	 * @since 8.4.0
+	 */
+	$theme_include_list               = apply_filters( 'woocommerce_hooked_blocks_theme_include_list', array( 'Twenty Twenty-Four', 'Twenty Twenty-Three', 'Twenty Twenty-Two', 'Tsubaki', 'Zaino', 'Thriving Artist', 'Amulet', 'Tazza' ) );
+	$should_set_hooked_blocks_version = in_array( $active_theme_name, $theme_include_list, true );
+
+	if ( $should_set_hooked_blocks_version && false === $option_value ) {
+		// Set 8.4.0 as the version for existing stores that are using a theme that supports the Block Hooks API.
+		// This will ensure that the Block Hooks API is enabled for these stores and works as expected.
+		// Existing stores that aren't running approved block themes will not have the Block Hooks API enabled.
+		add_option( $option_name, '8.4.0' );
+	} elseif ( ! $should_set_hooked_blocks_version && false === $option_value ) {
+		// For block themes that aren't approved themes set this option to "no" so we can disable the Block Hooks API.
+		// This will allow us to set a version when users switch from a classic theme to a block theme.
```

Comment:
Hmmm, but here ([line 2768](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-ef66873bc65699d07994f8ac9621c771df9f9e63ce74c94b70defac87aab52d1R2768)) we are setting the option to `no`. That means it won't be overridden when switching from a classic theme to a block theme (according to the check [here](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-d49488c34017d273db858b7b6eff58f22d95310991d26638e75793c6a1e98678R4060)), no? Thus, contradicting this part of the comment: `This will allow us to set a version when users switch from a classic theme to a block theme`.

I might be miunsdertanding this comment, but I think it doesn't match what's happening in this part of the `elseif` clause? :thinking:

Reply:
The first line of the function [checks if the user is _currently_ running a classic theme](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-ef66873bc65699d07994f8ac9621c771df9f9e63ce74c94b70defac87aab52d1R2737-R2739) and if they are we don't set an option _at all_ which means when they [switch from classic > block theme we can set one then](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-d49488c34017d273db858b7b6eff58f22d95310991d26638e75793c6a1e98678R4055).